api/odl.py

```diff
@@ -344,7 +345,8 @@ def checkout(self, patron, pin, licensepool, internal_format):
     def _checkout(self, patron_or_client, licensepool, hold=None):
         _db = Session.object_session(patron_or_client)
 
-        if licensepool.licenses_owned < 1:
+        unexpired_licenses_owned = len([license for license in licensepool.licenses if not license.is_expired])
+        if unexpired_licenses_owned < 1:
```

Comment:
I think it's possible to have thousands of distinct licenses (for a popular book), which could make this a performance problem. Can you either do this work in the database, or use `any` to bail out of the iteration the first time you find an unexpired title?

Reply:
Thanks for the insight. I'll look into this.